ravenframework/Models/PostProcessors/Validations/Representativity.py

```diff
@@ -0,0 +1,464 @@
+
+    for i,targ in enumerate(self.targetOutputs):
+      name3 = "CorrectedTargets_{}".format(targ.split("|")[-1])
+      outs[name3] = correctedTargets[i]
+      for j,feat in enumerate(self.prototypeOutputs):
+        name1 = "BiasFactor_Mock{}_Tar{}".format(feat.split("|")[-1], targ.split("|")[-1])
+        name2 = "ExactBiasFactor_Mock{}_Tar{}".format(feat.split("|")[-1], targ.split("|")[-1])
+        outs[name1] = r[i,j]
+        outs[name2] = rExact[i,j]
+      for k,tar in enumerate(self.targetOutputs):
+        if k == i:
+          name3 = "CorrectedVar_Tar{}".format(tar.split("|")[-1])
+          name4 = "ExactCorrectedVar_Tar{}".format(tar.split("|")[-1])
+        else:
+          name3 = "CorrectedCov_Tar{}_Tar{}".format(targ.split("|")[-1], tar.split("|")[-1])
+          name4 = "ExactCorrectedCov_Tar{}_Tar{}".format(targ.split("|")[-1], tar.split("|")[-1])
+        outs[name3] = UtarVarTilde[i,k]
+        outs[name4] = UtarVarTildeExact[i,k]
+    return outs
+
+  def _generateSensitivityMatrix(self, outputs, inputs, sensDict, datasets, normalize=True):
+    """
+      Reconstruct sensitivity matrix from the Basic Statistic calculation
+      @ In, inputs, list, list of input variables
+      @ In, outputs, list, list of output variables
+      @ In, sensDict, dict, dictionary contains the sensitivities
+      @ Out, sensMatr, numpy.array, 2-D array of the reconstructed sensitivity matrix
+    """
+    sensMatr = np.zeros((len(outputs), len(inputs)))
+    inputVars = [x.split("|")[-1] for x in inputs]
+    outputVars = [x.split("|")[-1] for x in outputs]
+    for i, outVar in enumerate(outputVars):
+      for j, inpVar in enumerate(inputVars):
+        senName = "{}_{}_{}".format(self.senPrefix, outVar, inpVar)
+        # Assume static data (PointSets are provided as input)
+        if not normalize:
+          sensMatr[i, j] = sensDict[senName][0]
+        else:
+          sensMatr[i, j] = sensDict[senName][0]* datasets[inpVar].meanValue / datasets[outVar].meanValue
+    return sensMatr
+
+  def _computeMoments(self, datasets, features, targets):
+    """
+      A utility function to compute moments, mean value, variance and covariance
+        @ In, datasets, xarray datasets, datasets containing prototype (mock) data and target data
+        @ In, features, names of feature variables: measurables
+        @ In, targets, names of target variables: figures of merit (FOMs)
+        @ out, datasets, xarray datasets, datasets after adding moments
+    """
+    for var in [x.split("|")[-1] for x in features + targets]: #datasets.data_vars
+      datasets[var].attrs['meanValue'] = np.mean(datasets[var].values)
+      for var2 in [x.split("|")[-1] for x in features + targets]:
+        if var == var2:
+          datasets[var2].attrs['var'] = np.var(datasets[var].values)
+        else:
+          datasets[var2].attrs['cov_'+str(var)] = np.cov(datasets[var2].values,datasets[var].values)
+    return datasets
```

Comment:
This function modifies the datasets, so the input dataset will be changed by it.

Reply:
I actually meant to do that. it adds to the dataset. But it seems you guys do not prefer that. So do I need to return that in a different variable or do I need to use `.copy()` inside as well?

Reply:
Returning a different variable would probably be more efficient that using copy.